### PR TITLE
chore(docs): fix up some of the generated descriptions of various shared config types

### PR DESCRIPTION
**Pull request by tobz**
This PR is an incremental chunk of work needed before moving over sources and sinks to the new machine-generated Cue documentation.

Generally, it's simply rewording and porting over user-facing descriptions of types/enums/etc to generate more user-friendly documentation for shared Rust types that show up often in sources/sinks.

### CODE REVIEW

File: lib/codecs/src/decoding/mod.rs [hunk 1]
```diff
@@ -216,26 +227,45 @@ impl tokio_util::codec::Decoder for Framer {
...
+#[configurable(metadata(docs::enum_tag_description = "The codec to use for decoding events."))]
 pub enum DeserializerConfig {
-    /// Configures the `BytesDeserializer`.
+    /// Uses the raw bytes as-is.
     Bytes,
-    /// Configures the `JsonDeserializer`.
+
+    /// Decodes the raw bytes as [JSON][json].
+    ///
+    /// [json]: https://en.wikipedia.org/wiki/JSON
```
Comment (spencergilbert) [inline lib/codecs/src/decoding/mod.rs:245]: Thinking out loud, we should probably standardize/have guidelines on what to link. Spec site vs wiki vs whatever..

File: lib/codecs/src/encoding/mod.rs [hunk 2]
```diff
@@ -182,46 +182,64 @@ impl tokio_util::codec::Encoder<()> for Framer {
...
-    /// JSON serialization.
+    /// Encodes an event as [JSON][json].
+    ///
+    /// [json]: https://en.wikipedia.org/wiki/JSON
     Json,
 
-    /// Logfmt serialization.
+    /// Encodes an event as a [logfmt][logfmt] message.
+    ///
+    /// [logfmt]:
```
Comment (spencergilbert) [inline lib/codecs/src/encoding/mod.rs:212]: Missing link?

**Review by spencergilbert — APPROVED** [review]

File: lib/codecs/src/encoding/mod.rs (hunk 2 above)
Comment (tobz) [reply lib/codecs/src/encoding/mod.rs:212]: Ooh, yes, good call. 👍🏻  Will fix.

File: lib/codecs/src/decoding/mod.rs (hunk 1 above)
Comment (tobz) [reply lib/codecs/src/decoding/mod.rs:245]: Yeah, that's a fair point. I'll go and find the most official link I reasonably can for these and update this stuff.

Comment (tobz) [reply lib/codecs/src/decoding/mod.rs]: What do you mean by the "variation" in "targets"? Like the quality/official-ness of a given link?

**Pull request merged**